### PR TITLE
Issue 3808 4.28 release candidate

### LINKED ISSUE #3808: Bugs list v4.27.2
### Bugs
**V4 Regression**
- [x] ~~1. [Wrong link in log in/sign up form](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1335868332)~~
- [x] [2. Settings button not working](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1335877761) (Alessandro)
- [x] [3. Selecting object with BIM card/through BIM card opened takes measurement instead](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1335943184) (Daniel)
- [x] [4. Sample_Project federation tree doesn't work and architectural model elements cannot be selected ](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1337379034) (Santiago)
- [x] do not give new users toy project (carmen) 

**V5 UI**
- [x] [5. V5 Multiline Text Field scrollbars broken in Firefox](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1337647547)  (This will not be fixed - Alessandro)
- [x] [6. Clicking 'Upload file to container' does not add the file to the uploads list](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1337656630) (Alessandro)
- [x] [7. The error text is obscured on custom tickets properties](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1339720476) (Daniel)
- [x] ~~[8. A required viewpoint should not require both viewpoint and screenshot](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1339732552)~~
- [x] [9. Long property names are not truncated](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1339774211) (Alessandro)
- [x] [10. App crashes when changing model with custom tickets open](https://github.com/3drepo/3drepo.io/issues/3808#issuecomment-1342555156) (Daniel)

### PR DESCRIPTION
**Pull request by carmenfan**
This fixes #3808

#### Bugs

- [x] [Measurement Card activate/deactivate features - dismiss button does not work](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1342917009) (Daneil)
- [x] [Measurement Card: Min distance is not working as expected](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1342931676) (Keif via [devops#449](https://github.com/3drepo/DevOps/issues/499))
- [x] [Measurement Card: Surface area is not working as expected](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1342938047) (Keif via [devops#449](https://github.com/3drepo/DevOps/issues/499))
- [x] [Custom Tickets fields not case sensitive](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1342939239) (Daniel)
- [x] [Viewpoint should update screenshot](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1343023485) (Santiago)
- [x] [Update screenshot in issues fails](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1343036557) (Alessandro)
- [x] Properties should be able to have the same name as long as they are presented in different cases e.g. Egg vs EGG (carmen)

### UI Oddities

~~- [ ] [Notifications number is positioned incorrectly](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346616241) v4.30~~
- [x] [Column title has dropped down in user permissions and project permissions](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346767076) v4.30 (Daniel)
- [x] [Swap custom tickets icon with filled version](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346781667) THIS RELEASE (Daniel)

~~- [ ] [Arrows on custom ticket detail are way too big](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346794087) v4.30~~

- [x] [Ticket shouldn't load with title field in focused state](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346796007) THIS RELEASE (Santiago)

~~- [ ] [Missing Safetibase icon inside Safetibase cards](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346797569) v4.30~~
~~- [ ] [Gaps between action buttons at the top of cards](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346801720) v4.30~~
~~- [ ] [Instant clash tab on compare tab should be on a single line](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346827337) v4.30~~

**all crossed out items have been moved to #3822 (except [this one](https://github.com/3drepo/3drepo.io/pull/3817#issuecomment-1346616241) which is addressed in #3778)**

#### Description
- Addresses all bugs recorded on the bug list

#### Test cases
- Bugs listed on the bug list should now be resolved
- All 4.28 features should function as expected

### CODE REVIEW

File: frontend/src/v4/modules/risks/risks.sagas.ts [hunk 1]
```diff
@@ -633,14 +633,14 @@ function toggleMeasurementListeners(enabled) {
 
 export function* setMeasureMode({ measureMode }) {
 	try {
+		if (!measureMode) {
+			yield Viewer.clearMeasureMode();
+			return
+		}
 		toggleMeasurementListeners(false);
```
Comment (sanmont3drepo) [inline frontend/src/v4/modules/risks/risks.sagas.ts:640]: Im worried about the listeners not being turned off here

**Review by sanmont3drepo — CHANGES_REQUESTED** [review]

File: frontend/src/v4/modules/measurements/measurements.sagas.ts [hunk 2]
```diff
@@ -56,14 +56,14 @@ const onMeasurementChanged = () => {
 
 export function* setMeasureMode({ mode }) {
 	try {
+		if (!mode) {
+			yield Viewer.clearMeasureMode();
+			return
+		}
 		toggleMeasurementListeners(false);
```
[the remark this answers is not in the document]
Comment (The-Daniel) [reply frontend/src/v4/modules/measurements/measurements.sagas.ts:63]: Good point. I don't think I saw that

File: frontend/src/v4/services/viewer/viewer.tsx [hunk 3]
```diff
@@ -454,11 +454,8 @@ export class ViewerService {
 		await this.isViewerReady();
 		this.measureModeLabels = labels;
 
-		if (this.measureMode) {
-			this.clearMeasureMode();
-			if (!mode) {
-				return;
-			}
+		if (!mode) {
```
[the remark this answers is not in the document]
Comment (The-Daniel) [reply frontend/src/v4/services/viewer/viewer.tsx:459]: I've readded it

**Comment by miadibe** [comment]
### Bug 1. Measurement Card activate/deactivate features

###Description 
Clicking on any measurement button (locate point, point to point, laser, min distance, surface area, custom surface area) activates the feature but does not deactivate it on second click

### Current behaviour 
Click once to activate the feature. Click again and button remains orange. 
bug1

### Expected Behaviour
Click once to activate the feature (orange). Click again to deactivate it (grey). 
expected1

**Comment by miadibe** [comment]
### Bug 2. Measurement Card:  Min distance is not working as expected

###Description 
- Distance measurement shows that it is being calculated for the correct object selections (highlighted) but is actually not recorded or shown in the UI
- What happens instead is a multi-selection on single click highlighting every selected object

### Current behaviour
bug2

### Expected behaviour
expected2

**Comment by mosment3DRepo** [comment]
### Viewpoint should update screenshot 

**Current Behaviour**
When I created a new viewpoint it does not create a screenshot, so it needs to be added separately

https://user-images.githubusercontent.com/62879582/206498327-2d77c696-ea7f-4cec-a087-208c32f54d5d.mp4

**Expected Behaviour**
As with V4, when we create a viewpoint we should take a screenshot of the Unity window and attach it as the image.

When the viewpoint is updated it should ask if you also want to update the image, as per V4

**Comment by Amantini1997** [comment]
### In properties, delete Viewpoint should delete viewpoint

**Current Behaviour**
In any custom tickets, in the properties section (so not other modules), clicking "edit viewpoint > delete viewpoint" does not delete the viewpoint

**Expected Behaviour**
In any custom tickets, in the properties section (so not other modules), clicking "edit viewpoint > delete viewpoint" should delete the viewpoint

**Comment by sanmont3drepo** [comment]
Updating a viewpoint screenshot doesnt work on the first try.
Repro steps: 
- Navigate to a federation in v5
- Open a issue with a viewpoint 
- Click on 'update viewpoint'
- Click on 'yes' after being asked if you want to create a new screenshot

What happens:
- The screenshot modal fails to appear
What should happen:
- The screenshot modal should appear

image

After failing if you try again the modal will appear.

**Review by Amantini1997 — DISMISSED** [review]

**Review by The-Daniel — DISMISSED** [review]

**Review by sanmont3drepo — APPROVED** [review]

**Comment by george-hadfield** [comment]
### Notifications number is positioned incorrectly

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Column title has dropped down in user permissions and project permissions

image

**Comment by george-hadfield** [comment]
### Swap custom tickets icon with filled version

Existing
image

Updated
image

Updated icon available here https://www.figma.com/file/3FjNEOdQbcHwzquVZxUqMl/3D-Repo?node-id=2011%3A15811&t=Z07108aC1WrBtM7R-1

**Comment by george-hadfield** [comment]
### Arrows on custom ticket detail are way too big

Build
image

Design
image

**Comment by george-hadfield** [comment]
### Ticket shouldn't load with title field in focused state

Build
image

**Comment by george-hadfield** [comment]
### Missing Safetibase icon inside Safetibase cards

image

**Comment by george-hadfield** [comment]
### Gaps between action buttons at the top of cards

image
 
Buttons should align to the right with a gap of 5px

**Comment by george-hadfield** [comment]
### Instant clash tab on compare tab should be on a single line

Build
image

**Comment by sanmont3drepo** [comment]
> ### Notifications number is positioned incorrectly
> Build image
> 
> Design image

This one will be fixed in https://github.com/3drepo/3drepo.io/pull/3783

**Comment by sanmont3drepo** [comment]
> ### Ticket shouldn't load with title field in focused state
> Build image

Are you sure about this one? I might be wrong think we changed it by yours or Matt's request (@Amantini1997 ?)

**Pull request merged**